### PR TITLE
rename embed job input param

### PR DESCRIPTION
**Pull request by lfayoux**
updates embed-jobs api spec

### CODE REVIEW

File: cohere/client.py [hunk 1]
```diff
@@ -1047,7 +1047,7 @@ def wait_for_cluster_job(
 
     def create_embed_job(
         self,
-        input_dataset: Union[str, BaseDataset],
+        dataset: Union[str, BaseDataset],
```
Comment (mkozakov) [inline cohere/client.py:1050]: Billy wants to remove the union type, keep the name consistent between api and sdk. Would be good to get his input

Comment (manoj-cohere) [reply cohere/client.py:1050]: https://github.com/cohere-ai/blobheart/pull/14120#discussion_r1451608283

**Review by manoj-cohere — APPROVED** [review]

File: cohere/client.py [hunk 2]
```diff
@@ -1047,42 +1047,31 @@ def wait_for_cluster_job(
...
-        text_field: Optional[str] = None,
-        input_type: Optional[str] = None,
     ) -> EmbedJob:
         """Create embed job.
 
         Args:
-            input_dataset (Union[str, BaseDataset]): Dataset or dataset id with text to embed.
-            name (Optional[str], optional): The name of the embed job. Defaults to None.
-            model (Optional[str], optional): The model ID to use for embedding the text. Defaults to None.
+            dataset_id (str): dataset id with text to embed.
```
Comment (mkozakov) [inline cohere/client.py:1059]:
```suggestion
            dataset_id (str): ID of the dataset of text to embed.
```

File: cohere/client.py [hunk 3]
```diff
@@ -1047,42 +1047,31 @@ def wait_for_cluster_job(
...
+            model (str): The model ID to use for embedding the text.
```
Comment (mkozakov) [inline cohere/client.py:1060]:
```suggestion
            model (str): ID of the model to use for embedding the text.
```

**Review by mkozakov — APPROVED** [review]

**Review by mkozakov — APPROVED** [review]

**Pull request merged**